### ISSUE TITLE
Contract attributes gives its output all the same span

### ISSUE THREAD

**Issue by jdonszelmann** [rust-lang/rust#147851]
It was reported in https://github.com/rust-lang/rust/issues/146808 that the following code ICEs:

```rust
#![core::contracts::requires]
#[allow{}]
fn main() {}
```

(compiler version for future reference)

```
rustc 1.92.0-nightly (f46475914 2025-10-17)
binary: rustc
commit-hash: f46475914de626785090a05ae037578aaa119fc8
commit-date: 2025-10-17
host: x86_64-unknown-linux-gnu
release: 1.92.0-nightly
LLVM version: 21.1.3
```

The reason it ICEs has little to do with contract attributes, and more to do with the fact that if two spans of a suggestion overlap, rust ICEs. This can be triggered in other ways.

However, contract attributes trigger this precisely because they cause their output to all have the same span. This I think is a bug on its own.

Assigning @celinval for now since she's doing contract attributes as far as I'm aware?
